### PR TITLE
[WIP]: refactor : Enable @typescript-eslint/no-explicit-any rule and fix 

### PR DESCRIPTION
**Pull request by maloun96**
### SUMMARY
Enable @typescript-eslint/no-explicit-any and resolve errors.
Still in progress

### CODE REVIEW

**Comment by junlincc** [comment]
can we change the PR title?

**Comment by rusackas** [comment]
Converted it to draft and gave it a WIP title. Hope that's OK :D 

Thanks for tackling this!

**Comment by rusackas** [comment]
Oh, and small PRs are always good... this thing is getting BIG!

**Comment by nytai** [comment]
Oh wow, didn't realize we were making this change. After a quick review I realized we have duplicate version of `SavedQuery` and `SavedQueryObject` floating around. Making a note to self to do a pass at type consolidation later. Trying to knockout all the any's in a single PR seems very ambitious. Maybe we can just warn at first and then fix all the warnings over a the next couple of PRs

File: superset-frontend/src/constants.ts [hunk 1]
```diff
@@ -19,3 +19,5 @@
 export const DATETIME_WITH_TIME_ZONE = 'YYYY-MM-DD HH:mm:ssZ';
 
 export const TIME_WITH_MS = 'HH:mm:ss.SSS';
+
+export type $anyType = any;
```
Comment (ktmud) [inline superset-frontend/src/constants.ts:23]: I've seen people calling this `$FIX_ME`, but I guess `anyType` also works.

**Comment by maloun96** [comment]
> Oh wow, didn't realize we were making this change. After a quick review I realized we have duplicate version of `SavedQuery` and `SavedQueryObject` floating around. Making a note to self to do a pass at type consolidation later. Trying to knockout all the `any`s in a single PR seems very ambitious. Maybe we can just warn at first and then fix all the warnings in follow up PRs, before actually switching to error

Sure, it's in progress, and turned it to error to better see all the "any". I will switch back to warning when it will be done?

**Comment by maloun96** [comment]
Should I continue working on  this PR, it will be a big PR if it's ok
@etr2460 @rusackas

**Comment by etr2460** [comment]
@maloun96 I would recommend we do this in 2 steps:
1. Add the lint rule and do a find-replace of `any` => `$anyType`/`$TSFixMe`. This would be almost all autogenned then, and wouldn't require much review as the first PR
2. Start replacing the `$anyType`/`$TSFixMe` in subsequent PRs, maybe limiting the PRs to touching 10-20 files each

**Comment by rusackas** [comment]
@maloun96 Totally agree with @etr2460 - if this PR has all the "any" replaced with "$anyType" (or $TSFixMe - I'm not passionate about what it's called), then we should just review/merge this one, and follow up on refactoring in smaller chunks.

**Comment by rusackas** [comment]
@maloun96 I'm guessing this was closed due to conflict/rebasing madness, but would you have any interest in rekindling it?

**Pull request closed without merging**